clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -5348,6 +5348,13 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
     if (Args.hasFlag(options::OPT_fsycl_esimd_force_stateless_mem,
                      options::OPT_fno_sycl_esimd_force_stateless_mem, false))
       CmdArgs.push_back("-fsycl-esimd-force-stateless-mem");
+
+    // Add device trait macros args to CmdArgs and SYCLTargetMacroArg.
+    const auto DeviceTraitMacrosArgs = TC.getDeviceTraitMacrosArgs();
+    for (const auto &Arg : DeviceTraitMacrosArgs) {
+      CmdArgs.push_back(Arg);
```

Comment:
This looks like it will add the macro twice for the host compilation - once here with the `CmdArgs` and again when `D.getSYCLTargetMacroArgs` is traversed.